charts/vela-core/templates/defwithtemplate/config-dex-connector.yaml

```diff
@@ -0,0 +1,65 @@
+# Code generated by KubeVela templates. DO NOT EDIT. Please edit the original cue file.
+# Definition source cue file: vela-templates/definitions/internal/config-dex-connector.cue
+apiVersion: core.oam.dev/v1beta1
+kind: ComponentDefinition
+metadata:
+  annotations:
+    custom.definition.oam.dev/name.config.oam.dev: Dex Connector
+    definition.oam.dev/description: Config information to authenticate Dex connectors
+  labels:
+    custom.definition.oam.dev/catalog.config.oam.dev: velacore-config
+    custom.definition.oam.dev/multi-cluster.config.oam.dev: "false"
+    custom.definition.oam.dev/type.config.oam.dev: dex-connector
+  name: config-dex-connector
+  namespace: {{ include "systemDefinitionNamespace" . }}
+spec:
+  schematic:
+    cue:
+      template: |
+        output: {
+        	apiVersion: "v1"
+        	kind:       "Secret"
+        	metadata: {
+        		name:      parameter.name
+        		namespace: "vela-system"
+        		labels: {
+        			"config.oam.dev/catalog":       "velacore-config"
+        			"config.oam.dev/type":          "dex-connector"
+        			"config.oam.dev/multi-cluster": "false"
+        			"config.oam.dev/identifier":    parameter.name
+        		}
+        	}
+        	type: "Opaque"
+
+        	if parameter.github != _|_ {
+        		stringData: parameter.github
+        	}
+        	if parameter.ldap != _|_ {
+        		stringData: parameter.ldap
+        	}
+        }
+        parameter: {
```

Comment:
The definition is incomplete? there are other types. 

Advice to add a type parameter, it is better for user interactive configuration.

Reply:
Let's support the most frequently used ones at first. We can add one at any time after it's been verified.

`github` and `ldap` are types.

Reply:
a specified type can make it clear that users can only use one of the type. What if we both configure github and ldap here? That will cause conflict.

Reply:
Can you write a simple sample for the `type`? I didn't understand how to connect the `type` field with a `github`, `ldap` type and its propties.

Reply:
refer to https://github.com/oam-dev/kubevela/blob/master/charts/vela-core/templates/defwithtemplate/task.yaml#L173-L197

Reply:
@wonderflow Here is a sample from @barnettZQG .

```
repoType: oss | https | ssh
oss?: {
        bucket:   string
        endpoint: string
}
https?: {
      url:      string
      username: string
      password: string
}
// +usage=https://fluxcd.io/legacy/helm-operator/helmrelease-guide/chart-sources/#ssh
ssh?: {
    url:      string
    identity: string
}
```

Is it what your mean?

Reply:
@zzxwill yes, you should add `if` condition in template to check if the repoType exist like this https://github.com/oam-dev/kubevela/blob/master/charts/vela-core/templates/defwithtemplate/task.yaml#L72-L95